helpers.py

```diff
@@ -0,0 +1,89 @@
+import os
+import torch
+import shutil
+from faster_whisper import WhisperModel
+
+
+def vocal_extraction(audio_path, extract=True):
+    if extract:
+        os.system(
+            f'python3 -m demucs.separate -n htdemucs --two-stems=vocals "{audio_path}" -o "temp_outputs"'
+        )
+        vocal_target = os.path.join(
+            "temp_outputs",
+            "htdemucs",
+            os.path.splitext(os.path.basename(audio_path))[0],
+            "vocals.wav",
+        )
+    return vocal_target
+
+
+# READ, MAP TIMESTAMPS AND REFORMAT FOR READIBILITY
+def text_speaker_map(whisper_results, temp_path):  # add whisper results here?
+    spk_tp = []
+    with open(os.path.join(temp_path, "pred_rttms", "mono_file.rttm"), "r") as f:
+        lines = f.readlines()
+        for line in lines:
+            line_list = line.split(" ")
+            s = int(float(line_list[5]) * 1000)
+            e = s + int(float(line_list[8]) * 1000)
+            spk_tp.append([s, e, int(line_list[11].split("_")[-1])])
+
+    with open("speaker_transcript.txt", "w", encoding="utf-8-sig") as f:
+        for i, seg in enumerate(whisper_results):
+            start = seg["start"]
+            end = seg["end"]
+            text = seg["text"]
+            spk = spk_tp[i][2]
+            # s = 'Speaker_'+str(spk), start, "-", str(end)+":", text
+            # print('Speaker_'+str(spk), start, "-", str(end)+":", text)
+            f.write(
+                "Speaker_"
+                + str(spk)
+                + " "
+                + str(start)
+                + "-"
+                + str(end)
+                + ":"
+                + str(text)
+                + "\n"
+            )
+    return spk_tp
+
+
+# this function won't be needed once ran on the server
+def transcribe(
+    audio_file: str, language: str, model_name: str, compute_dtype: str, device: str
+):
+    # Faster Whisper non-batched
+    # Run on GPU with FP16
+    whisper_model = WhisperModel(model_name, device=device, compute_type=compute_dtype)
+
+    # or run on GPU with INT8
+    # model = WhisperModel(model_size, device="cuda", compute_type="int8_float16")
+    # or run on CPU with INT8
+    # model = WhisperModel(model_size, device="cpu", compute_type="int8")
+
+    segments, info = whisper_model.transcribe(
+        audio_file,
+        language=language,
+        beam_size=5,
+        word_timestamps=True,
+        vad_filter=True,
+    )
+    whisper_results = []
+    for segment in segments:
+        whisper_results.append(segment._asdict())
+    # clear gpu vram
+    del whisper_model
+    torch.cuda.empty_cache()
+    return whisper_results, language
```

Comment:
I think the idea, in the end, was to use Whisper transcription as input.

In that case, move anything not related to the core diarization to a separate python module called generate_input.py or something like that, to make it easy to distinguish stuff that should and should not be part of this worker. This allows someone working on this repo to just generate input once, and not run the entire Whisper model when they want to run/test the core of this worker. 

Or perhaps leave it out completely, and just add one set of example input (transcript, audio, ...) for a piece of open domain material.

Reply:
I have put the transcribe function in another file, I will state on the README that this should not be part of the final worker.